app/gui2/src/components/GraphEditor.vue

```diff
@@ -266,12 +266,29 @@ const graphBindingsHandler = graphBindings.handler({
       if (currentMethodName == null) {
         bail(`Cannot get the method name for the current execution stack item. ${currentMethod}`)
       }
-      graphStore.editAst((module) => {
-        if (graphStore.moduleRoot == null) bail(`Module root is missing.`)
-        const topLevel = module.get(graphStore.moduleRoot)
-        assert(topLevel instanceof BodyBlock)
-        return performCollapse(info, module, topLevel, graphStore.db, currentMethodName)
-      })
+      const currentFunctionEnv = environmentForNodes(selected.values())
```

Comment:
Having this be in a closure was a good thing, though now it should rather be `graphStore.editScope`. You can also remove the `editAst` function if it is now unused.

Reply:
That would be nice, but unfortunately it results in an extremely ugly code. I need to set a few variables outside of the edit’s scope for setting correct positions, but these variables are only available after collapsing. So I’d prefer leaving it as it is for now.